components/eam/src/physics/yaml/nucleate_ice_cam/f90_yaml/nucleate_ice_cam_calc_beg_yml.f90

```diff
@@ -0,0 +1,101 @@
+
+        yaml%lchnk_print = lchnk !(**remove these if generating data for a dependent subroutines**)
+        yaml%flag_print  = .true.!(**remove these if generating data for a dependent subroutines**)
+
+        !open I/O yaml files (it can have an extra optional argument to pass a unique string to differentiate file names)
+        call open_files('nucleate_ice_cam_calc', &  !intent-in
+             unit_input, unit_output) !intent-out
+!             unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print,yaml%col_print, &
+             'compute_tendencies',yaml%nstep_print, yaml%lev_print)
+
+        !< add code for writing data here>
+        call write_var(unit_input, unit_output, 'modeptr_accum', modeptr_accum)
+        call write_var(unit_input, unit_output, 'modeptr_aitken', modeptr_aitken)
+        call write_var(unit_input, unit_output, 'modeptr_coarse', modeptr_coarse)
+        call write_1d_var(unit_input, unit_output, 'numptr_amode', ntot_amode, numptr_amode(:))         
+        call write_1d_var(unit_input, unit_output, 'alnsg_amode', ntot_amode, alnsg_amode(:))
+        
+        call write_1d_var(unit_input, unit_output, 'lptr_dust_a_amode', ntot_amode, lptr_dust_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_nacl_a_amode', ntot_amode, lptr_nacl_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_so4_a_amode', ntot_amode, lptr_so4_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_bc_a_amode', ntot_amode, lptr_bc_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_pom_a_amode', ntot_amode, lptr_pom_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_soa_a_amode', ntot_amode, lptr_soa_a_amode(:))
+        call write_1d_var(unit_input, unit_output, 'lptr_mom_a_amode', ntot_amode, lptr_mom_a_amode(:))
+
+        call write_var(unit_input, unit_output,'ncol',ncol)
+        call write_var(unit_input, unit_output,'lchnk',lchnk)
+        call write_var(unit_input, unit_output,'temperature',temperature(yaml%col_print, yaml%lev_print))      
+        call write_1d_var(unit_input, unit_output,'state_q', pcnst,state_q(yaml%col_print, yaml%lev_print,:))
+        call write_var(unit_input, unit_output,'pmid',pmid(yaml%col_print, yaml%lev_print))
+        call write_var(unit_input, unit_output,'rho',rho(yaml%col_print, yaml%lev_print))
+        call write_var(unit_input, unit_output,'wsubi',wsubi(yaml%col_print, yaml%lev_print))
+        call write_var(unit_input, unit_output,'ast',ast(yaml%col_print, yaml%lev_print))
+        call write_1d_var(unit_input, unit_output,'dgnum', ntot_amode,dgnum(yaml%col_print, yaml%lev_print,:))        
+
```

Comment:
May you write so4_sz_thresh_icenuc? see line 483

Reply:
added to the input list